src/server/frontend_wayland/session_lock_v1.cpp

```diff
@@ -241,6 +250,7 @@ void mf::SessionLockV1::unlock_and_destroy()
 {
     if (!manager.try_unlock(this))
     {
+        mir::log_info("SessionLockV1 attempted to unlock_and_destroy but it is not allowed to");
         BOOST_THROW_EXCEPTION(mw::ProtocolError(
             resource,
             Error::invalid_unlock,
```

Comment:
This would happen in the generator, right?

Reply:
I've not looked. But, in general terms, if we throw an exception there should somewhere we catch it. And the catch site has more context about what was intended to happen and how to handle the failure (including what needs logging)

Reply:
Yeah we `catch(ProtocolError const& err)` everywhere. I can either make a ticket to add a log to these statements in the generator code OR `WAYLAND_DEBUG=1` would just give you that info anyway, since we forward all of those errors with `wl_resource_post_error(...)`

Reply:
As you imply, reporting the error with both "post_error" and logging sounds redundant

Reply:
Cool, then I'll leave it for now :+1:

Reply:
> As you imply, reporting the error with both "post_error" and logging sounds redundant

The reason we may want to log something as well as `wl_resource_post_error` is that `post_error` is otherwise silent in the *Mir* log (it's noisy on the cilent end; the client will be killed and will by default print the error we sent it).

“We deliberately killed a client because it behaved incorrectly” seems likely to be both infrequent and of interest to users when it happens.